File: examples/advanced-crud/src/main/java/org/example/advancedcrud/app/repository/MovieRepository.java
package org.example.advancedcrud.app.repository;

import org.example.advancedcrud.app.entity.Movie;
import org.example.advancedcrud.utils.BaseRepository;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MovieRepository extends JpaRepository<Movie, Long>, JpaSpecificationExecutor<Movie>, BaseRepository {

}

File: examples/basic-crud-kotlin/src/main/kotlin/org/example/basiccrudkotlin/app/repository/MovieRepository.kt
package org.example.basiccrudkotlin.app.repository;

import org.example.basiccrudkotlin.app.entity.Movie
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.rest.core.annotation.RepositoryRestResource

@RepositoryRestResource(path = "movies")
interface MovieRepository : JpaRepository<Movie, Long>, JpaSpecificationExecutor<Movie> {
}